tools/fm-editor/FeatureModelEditor.cpp

```diff
@@ -0,0 +1,290 @@
+  TreeView->setModel(TreeModel.get());
+  TreeView->setContextMenuPolicy(Qt::CustomContextMenu);
+  connect(TreeView, SIGNAL(customContextMenuRequested(const QPoint &)), this,
+          SLOT(createTreeContextMenu(const QPoint &)));
+}
+
+/// Build the graph view
+void FeatureModelEditor::buildGraph() {
+  Graph = std::make_unique<FeatureModelGraph>(FeatureModel.get());
+  for (auto &Node : *Graph->getNodes()) {
+    QObject::connect(Node.get(), &FeatureNode::clicked, this,
+                     &FeatureModelEditor::loadFeature);
+    QObject::connect(Node.get(), &FeatureNode::inspectSource, this,
+                     &FeatureModelEditor::inspectFeatureSources);
+  }
+}
+void FeatureModelEditor::featureAddDialog() { featureAddDialogChild(nullptr); }
+/// Spawn a Dialog to select data to add a Feature
+void FeatureModelEditor::featureAddDialogChild(Feature *ParentFeature) {
+  FeatureAddDialog AddDialog(Graph.get(), this, ParentFeature);
+  if (AddDialog.exec() == QDialog::Accepted) {
+    Feature *Parent =
+        FeatureModel->getFeature(AddDialog.getParent().toStdString());
+    auto NewFeature = AddDialog.getFeature();
+
+    auto *NewNode = Graph->addNode(NewFeature.get(),
+                                   Graph->getNode(Parent->getName().str()));
+    auto *NewTreeItem =
+        TreeModel->addFeature(NewFeature.get(), Parent->getName().str());
+    connect(NewTreeItem, &FeatureTreeItem::inspectSource, this,
+            &FeatureModelEditor::inspectFeatureSources);
+    connect(NewNode, &FeatureNode::clicked, this,
+            &FeatureModelEditor::loadFeature);
+    connect(NewNode, &FeatureNode::inspectSource, this,
+            &FeatureModelEditor::inspectFeatureSources);
+    auto Transaction = vara::feature::FeatureModelTransaction<
+        vara::feature::detail::ModifyTransactionMode>::
+        openTransaction(*FeatureModel);
+    auto PotentialRelations =
+        Parent->getChildren<vara::feature::Relationship>(1);
+    if (!PotentialRelations.empty()) {
+      Transaction.addFeature(std::move(NewFeature),
+                             *PotentialRelations.begin());
+    } else {
+      Transaction.addFeature(std::move(NewFeature), Parent);
+    }
+    Transaction.commit();
+  }
+}
+
+void FeatureModelEditor::removeFeature(bool Recursive, Feature *Feature) {
+  TreeModel->deleteFeatureItem(Recursive, Feature);
+  Graph->deleteNode(Recursive, Feature);
+  Ui->featureInfo->clear();
+  vara::feature::removeFeature(*FeatureModel, Feature, Recursive);
+}
+
+/// Save the current State of the Feature Model
+void FeatureModelEditor::save() {
+  SavePath = QFileDialog::getSaveFileName(this, tr("Save File"), ModelPath,
+                                          tr("XML files (*.xml)"));
+  vara::feature::FeatureModelXmlWriter FMWrite{*FeatureModel};
+  FMWrite.writeFeatureModel(SavePath.toStdString());
+}
+
+/// Loead the source files of the Feature to be selectable by the user and set
+/// the Feature as CurrentFeature. \param Feature
```

Comment:
```suggestion
/// Lead the source files of the Feature to be selectable by the user and set
/// the Feature as CurrentFeature.
///
/// \param Feature
```
You should also add a fitting description to the parameter

PS: the other interface methods could use some parameter comments, too.